src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -1,751 +1,5624 @@
+        /// <param name="dimensions"><see cref="ReadOnlySpan{T}"/> with the new axis ordering.</param>
+        public static Tensor<T> PermuteDimensions<T>(this Tensor<T> tensor, ReadOnlySpan<int> dimensions)
+        {
+            if (tensor.Rank == 1)
+            {
+                return tensor;
+            }
+            else
+            {
+                scoped Span<nint> newLengths = TensorOperation.RentedBuffer.CreateUninitialized(tensor.Rank, out TensorOperation.RentedBuffer<nint> lengthsRentedBuffer);
+                scoped Span<nint> newStrides = TensorOperation.RentedBuffer.CreateUninitialized(tensor.Rank, out TensorOperation.RentedBuffer<nint> stridesRentedBuffer);
+                scoped Span<int> newLinearOrder = TensorOperation.RentedBuffer.CreateUninitialized(tensor.Rank, out TensorOperation.RentedBuffer<int> linearOrderRentedBuffer);
+
+                Tensor<T> outTensor;
+
+                if (dimensions.IsEmpty)
+                {
+                    for (int i = 0; i < tensor.Rank; i++)
+                    {
+                        newLengths[i] = tensor.Lengths[tensor.Rank - 1 - i];
+                        newStrides[i] = tensor.Strides[tensor.Rank - 1 - i];
+                        newLinearOrder[i] = tensor._shape.LinearRankOrder[tensor.Rank - 1 - i];
+                    }
+                }
+                else
+                {
+                    if (dimensions.Length != tensor.Lengths.Length)
+                        ThrowHelper.ThrowArgument_PermuteAxisOrder();
+
+                    for (int i = 0; i < dimensions.Length; i++)
+                    {
+                        newLengths[i] = tensor.Lengths[dimensions[i]];
+                        newStrides[i] = tensor.Strides[dimensions[i]];
+                        newLinearOrder[i] = tensor._shape.LinearRankOrder[dimensions[i]];
+                    }
+                }
+                outTensor = new Tensor<T>(tensor._values, tensor._start, newLengths, newStrides, newLinearOrder);
+
+                lengthsRentedBuffer.Dispose();
+                stridesRentedBuffer.Dispose();
+                linearOrderRentedBuffer.Dispose();
+
+                return outTensor;
+            }
+        }
+        #endregion
+
+        #region Reshape
+        /// <summary>
+        /// Reshapes the <paramref name="tensor"/> tensor to the specified <paramref name="lengths"/>. If one of the lengths is -1, it will be calculated automatically.
+        /// Does not change the length of the underlying memory nor does it allocate new memory. If the new shape is not compatible with the old shape,
+        /// an exception is thrown.
+        /// </summary>
+        /// <param name="tensor"><see cref="Tensor{T}"/> you want to reshape.</param>
+        /// <param name="lengths"><see cref="ReadOnlySpan{T}"/> with the new dimensions.</param>
+        public static Tensor<T> Reshape<T>(this Tensor<T> tensor, ReadOnlySpan<nint> lengths)
```

Comment:
This method is almost the same as the one using ReadOnlyTensor. Is it possible to create ReadOnlyTensor and call the other one? or will be overhead there?

Reply:
The problem is that right now there is no way to convert a TensorSpan/ReadOnlyTensorSpan to a Tensor. You can go from a Tensor to them, but not the reverse, so there is no good way, currently, of doing what you are suggesting. 

We have talked about it though and plan on taking an API for that through api review, it just doesn't exist currently.